### PR TITLE
Rework roles to use better defaults

### PR DESCRIPTION
**Pull request by jwflory**
This commit is sizeable and makes many changes to various parts of the WordPress installation process. This change is explained below:

common
------

* Remove existing repo configuration
* Install EPEL via package
* Use IUS repositories for newer upstream versions

mariadb
-------

* Use MariaDB 10.1+
* Remove variables (hard-codes a port, but default is okay anyways)

nginx
-----

* Minor changes / clean-up
* Rework template for wordpress server block configuration

php-fpm
-------

* Use PHP 7.2+

Other notes
-----------

* Everything tested up to WordPress role - WordPress role needs work

@nlMeminger Could you review? Also tagging @Qalthos to take him up on his offer to review a PR here and there, since we're just getting started. :slightly_smiling_face:

### CODE REVIEW

File: roles/mariadb/tasks/main.yml [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+- name: install MariaDB
+  package: name={{ item }} state=installed
+  with_items:
+   - mariadb101u-server
 
-- name: Create Mysql configuration file
-  template: src=my.cnf.j2 dest=/etc/my.cnf
+- name: copy MySQL configuration file
+  template:
+    src: my.cnf.j2
```
Comment (Tjzabel) [inline roles/mariadb/tasks/main.yml]: yay for jinja2 templates

File: roles/mariadb/tasks/main.yml [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
 
-- name: Create MariaDB log file
-  file: path=/var/log/mysqld.log state=touch owner=mysql group=mysql mode=0775
+- name: create MariaDB log file
+  file:
+    path: /var/log/mysqld.log
+    state: touch
+    owner: mysql
+    group: mysql
+    mode: 0775
```
Comment (Tjzabel) [inline roles/mariadb/tasks/main.yml]: Do you want 755 or 644 for the log file? I'm not sure what permissions this specific log file needs.

File: roles/wordpress-stack-install.yml [hunk 3]
```diff
@@ -1,10 +1,8 @@
 ---
 - name: Install WordPress, MariaDB, Nginx, and PHP-FPM
   hosts: wordpress-server
-  remote_user: root
-  # remote_user: user
-  # sudo: yes
-
+  become: yes
```
Comment (axk4545) [inline roles/wordpress-stack-install.yml]: These two lines seem off to me...

Comment (axk4545) [reply roles/wordpress-stack-install.yml]: @jwflory Got it. You don't need `become_user` and it should be `become: true` in the latest versions of Ansible

Comment (Tjzabel) [reply roles/wordpress-stack-install.yml]: Which two lines? It actually is correct to use both `become` and `become_user`

File: roles/wordpress-stack-install.yml [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
+  become_user: root
```
Comment (Tjzabel) [inline roles/wordpress-stack-install.yml]: I feel like we are definitely making a bigger deal of it than it needs to be, but it's also good to know proper etiquette. It seems that when you use `hosts:`, you should generally follow it with a `remote_user`. For instance:
```
hosts: wordpress-server
remote_user: <regular user>
become: yes # become root
```
When you wish to change to the `root` user, only `become: yes` is necessary.

File: roles/mariadb/tasks/main.yml [hunk 5]
```diff
@@ -1,28 +1,40 @@
...
-- name: Install MariaDB package
-  yum: name={{ item }} state=installed
+- name: remove older versions of conflicting packages
+  package: name={{ item }} state=absent
   with_items:
-   - mariadb-server
-   - MySQL-python
-   - libselinux-python
-   - libsemanage-python
+    - mariadb-libs
```
[the remark this answers is not in the document]
Comment (jwflory) [reply roles/mariadb/tasks/main.yml]: @Qalthos Thanks! I didn't know this. :slightly_smiling_face:

**Comment by jwflory** [comment]
I rebased my commits from everyone's feedback. Thanks everyone for the thorough review! Merging. :clapper:

**Pull request merged**